crates/driver/src/infra/solver/dto/auction.rs

```diff
@@ -244,16 +275,19 @@ struct WeightedProductPool {
     address: eth::H160,
     #[serde_as(as = "serialize::U256")]
     gas_estimate: eth::U256,
-    tokens: HashMap<eth::H160, WeightedProductReserve>,
+    tokens: IndexMap<eth::H160, WeightedProductReserve>,
```

Comment:
We use an IndexMap here since Balancer tokens are also ordered at the smart contract level.